src/main/scala/nl.knaw.dans.easy.umd/Transformer.scala

```diff
@@ -49,13 +50,35 @@ object Transformer {
             Success(())
         }
       case ("EMD", "orgISNI") =>
-        if ((oldXML \\ "organization").exists(_.text == expectedOldValue))  Success(())
+        if ((oldXML \\ "organization").exists(_.text == expectedOldValue)) Success(())
         else Failure(new NotImplementedError(s"no organization with name [$expectedOldValue] found."))
       case _ =>
         Success(())
     }
   }
 
+  private def addChildTransformer(label: String, newChild: Node): RuleTransformer =
+    new RuleTransformer(new RewriteRule {
+      override def transform(n: Node): Seq[Node] = n match {
+        case n @ Elem(_, `label`, _, _, _*) => addChild(n, newChild)
+        case other => other
+      }
+    })
+
+  private def addChild(n: Node, newChild: Node) = n match {
+    case Elem(prefix, label, attribs, scope, child @ _*) if !childExists(child, newChild) =>
+      Elem(prefix, label, attribs, scope, false, child ++ newChild: _*)
+    case other => other
+  }
+
+  private def childExists(child: Seq[Node], newChild: Node): Boolean = {
+    (child contains newChild) || {
+      val childS = child.foldLeft("")(_ + _.toString.replaceAll("\\s+", ""))
+      val newChildS = newChild.foldLeft("")(_ + _.toString.replaceAll("\\s+", ""))
+      childS contains newChildS
```

Comment:
I'm not sure if this is the best way to go... Perhaps @jo-pol can give you some pointers here?

Reply:
Not sure what @rvanheest means but you are repeating yourself. Press ALT-Enter at the underscore:
![image](https://user-images.githubusercontent.com/10553630/82306979-0c753580-99c0-11ea-94cd-22bb5aeb9a61.png)
subsequently you can extract the method, Intellij automatically recognizes the duplicate.

But I'm confused, `child` and `newChild` have different types. Why apply the same action? What is the purpose of the replacement anyway?

I also very much doubt toString is what you want. You might want to treat different indentations as equal. Look in the EASY code base for the use of PrettyPrinter to fix the indentation. But then you may still have a problem with attributes in different order that you want to consider equal.

Reply:
I don't understand, where am I repeating myself?
Nothing is replaced here, but checked if `child` and `newChild` are exactly the same (after stripping all white spaces).  So, toString is what I want here.
When `newChild` is a simple element, then the test `child contains newChild` already  returns true if they are the same, otherwise we have to indeed stringify the nodes and compare the strings.